modules/dnn_superres/src/dnn_superres_quality.cpp

```diff
@@ -0,0 +1,470 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+
+#include "opencv2/dnn_superres_quality.hpp"
+
+namespace cv
+{
+    namespace dnn_superres
+    {
+        int DnnSuperResQuality::fontFace = cv::FONT_HERSHEY_COMPLEX_SMALL;
+
+        double DnnSuperResQuality::fontScale = 1.0;
+
+        Scalar DnnSuperResQuality::fontColor = Scalar(255,255,255);
+
+        void DnnSuperResQuality::setFontFace(int fontface)
+        {
+            fontFace = fontface;
+        }
+
+        void DnnSuperResQuality::setFontScale(double fontscale)
+        {
+            fontScale = fontscale;
+        }
+
+        void DnnSuperResQuality::setFontColor(cv::Scalar fontcolor)
+        {
+            fontColor = fontcolor;
+        }
+
+        double DnnSuperResQuality::psnr(Mat img, Mat orig)
+        {
+            CV_Assert(!img.empty());
+            CV_Assert(!orig.empty());
+            CV_Assert(img.type() == CV_8UC3);
+            CV_Assert(img.type() == orig.type());
+
+            Mat img_float;
+            img.convertTo(img_float, CV_32F);
+
+            Mat orig_float;
+            orig.convertTo(orig_float, CV_32F);
+
+            Mat img_diff;
+            absdiff(orig_float, img_float, img_diff);
+
+            Mat img_mul = img_diff.mul(img_diff);
+
+            Scalar sum = cv::sum(img_mul);
+
+            double rgb_sum = 0.0;
+            for(int i = 0; i < 4; i++){
+                rgb_sum += sum[i];
+            }
+
+            double mse = rgb_sum / (double) (3.0 * img_float.total());
+
+            double max = 255 * 255;
+            double psnr = 10 * log10(max / (double) mse);
+
+            return psnr;
+        }
+
+        double DnnSuperResQuality::ssim(Mat img, Mat orig)
+        {
```

Comment:
Did you take a look on SSIM from `quality` module (opencv_contrib)?

Reply:
Replaced it with using quality module implementation.